opensearch_py_ml/sentence_transformer_model/sentencetransformermodel.py

```diff
@@ -743,6 +787,91 @@ def set_up_accelerate_config(
         with open(file_path, "w") as file:
             yaml.dump(default_file, file)
 
+    def make_model_config_json(
+        self,
+        model_name: str = None,
+        version_number: int = 1,
+        embedding_dimension: int = 768,
+        all_config: str = None,
+        model_type: str = None,
+    ) -> None:
+        """
+        parse from config.json file of pre-trained hugging-face model to generate a model_config.json file. If all required
+        fields are given by users, use the given parameters and will skip reading the config.json,
+
+        Parameters
+        ----------
+        model_name: str = None
+            Optional, The name of the model. If None, default to parse from model id, for example,
+            'msmarco-distilbert-base-tas-b'
+        version_number: int = 1,
+            Optional, The version number of the model. If None, default to be 1.
+        embedding_dimension: int = 768,
+            Optional, the embedding_dimension of the model. If None, parse embedding_dimension from the config file of
+             pre-trained hugging-face model, if not found, default to be 768.
+        all_config: dict = None,
+            Optional, the all_config of the model. If None, parse all contents from the config file of pre-trained
+            hugging-face model.
+        model_type: str = None,
+            Optional, the model_type of the model. If None, parse model_type from the config file of pre-trained
+            hugging-face model.
+
+        Returns
+        -------
+        None
+        """
+        folder_path = self.folder_path
+        config_json_file_path = os.path.join(folder_path, "config.json")
+        if model_name is None:
+            model_name = self.model_id.split("/")[-1]
+
+        if all_config is None or model_type is None or embedding_dimension == 768:
+            if not os.path.exists(config_json_file_path):
+                raise Exception(
+                    str(
+                        "Cannot find config.json in"
+                        + config_json_file_path
+                        + ". Please check the config.son file in the path."
+                    )
+                )
+            f = open(config_json_file_path)
+            if all_config is None:
+                all_config = json.load(f)
+            if model_type is None:
+                model_type = all_config["model_type"]
```

Comment:
We should check if `all_config` has a index `model_type`. If not then this code will throw an exception

Reply:
all config fields is optional, but I added the check for "model_type".